cloud/ec2_ami_copy.py

```
@@ -0,0 +1,224 @@
+    description:
+      - the source region that AMI should be copied from
+    required: true
+    default: null
+    aliases: []
+  dest_region:
+    description:
+      - the destination region that AMI should be copied to
+    required: true
+    default: null
+    aliases: []
+  source_image_id:
+    description:
+      - the id of the image in source region that should be copied
+    required: true
+    default: null
+    aliases: []
+  name:
+    description:
+      - The name of the new image to copy
+    required: false
+    default: null
+    aliases: []
+  description:
+    description:
+      - An optional human-readable string describing the contents and purpose of the new AMI.
+    required: false
+    default: null
+    aliases: []
+  wait:
+    description:
+      - wait for the copied AMI to be in state 'available' before returning.
+    required: false
+    default: "no"
+    choices: [ "yes", "no" ]
+    aliases: []
+  wait_timeout:
+    description:
+      - how long before wait gives up, in seconds
+    required: false
+    default: 1200
+    aliases: []
+  tags:
+    description:
+      - a hash/dictionary of tags to add to the new copied AMI; '{"key":"value"}' and '{"key":"value","key":"value"}'
+    required: false
+    default: null
+    aliases: []
+
+author: Amir Moulavi <amir.moulavi@gmail.com>
+extends_documentation_fragment: aws
+'''
+
+EXAMPLES = '''
+# Basic AMI Copy
+- local_action:
+    module: ec2_ami_copy
+    aws_access_key: "-----REMOVED-------"
+    aws_secret_key: "-----REMOVED-------"
+    source_region: eu-west-1
+    dest_region: us-east-1
+    source_image_id: ami-xxxxxxx
+    name: SuperService-new-AMI
+    description: latest patch
+    tags: '{"Name":"SuperService-new-AMI", "type":"SuperService"}'
+    wait: yes
+  register: image_id
+'''
+
+
+import sys
+import time
+
+try:
+    import boto
+    import boto.ec2
+except ImportError:
+    print "failed=True msg='boto required for this module'"
+    sys.exit(1)
```

Instead of calling `sys.exit` it's better to return a failure to Ansible:

    try:
      import boto
      import boto.ec2
      HAS_BOTO = True
    except ImportError:
      HAS_BOTO = False

    if not HAS_BOTO:
        module.fail_json(msg="boto required for this module")